proposals/0051-serialized-manifest.md

```diff
@@ -0,0 +1,171 @@
+- Expose all information coming from the configuration.
+
+# Detailed Design
+
+> [!IMPORTANT]
+> The proposal will use some paths/names that might not be official, only to provide some scenarios or examples. Please refer to [#proposed-apis] to understand the final and proposed modules.  
+
+The main idea is to provide a virtual module with sub-paths (e.g. `astro:manifest/client`, `astro:manifest/server`), each sub path will expose information that won't be available in the other sub paths, so information won't be repeated. These modules will be compiled into code, so a piece information e.g. `trailingSlash` should be available only once in order to avoid polluting the final bundle with repeated code.
+
+Some sub-paths can't be used inside client-side code, because they could expose sensitive information, such as file system paths. Hence, we should divide the information based on where they could be used. We can't track if binding e.g. `config.srcDir` is used inside a client script, but we can track if a `import { srcDir } from astro:manifest/paths` is used inside a client side code. 
+
+## Use the manifest
+
+Internally, we already have this information, which is the `SSRManifest`, **not** the `AstroConfig`. The `AstroConfig` disappears after the build, instead we have
+the `SSRManifest` that is created during the build, and passed to the Astro application even when it has on-demand pages.
+
+The source code already does a very good job to serialise the information across pipelines (dev, SSG, SSR and container) and make sure that all of them contain the same information. Many of this information are computed from the user configuration.
+
+## What to expose
+
+As a rule of thumb, the information to expose should adhere to the following guidelines:
+- Information that can be serialised. Functions, `Map`, `Set`, etc. shouldn't be exposed. As long as a field can go through `JSON.strigify`, it's a good candidate. 
+- Information that can be used directly by our end-users. Integration developers have access to `AstroConfig`.
+- Information that isn't present inside `AstroConfig`. As mentioned previously, the `SSRManifest` contains useful information that are often used by on-demand pages and adapters.
+  An example is `SSRManifest.i18n.domainLookupTable`, which is used by adapters to understand the mapping of a domain to a locale.
+
+These are guidelines, so exceptions are possible, as long as they have a good reason to be exposed. 
+
+## Proposed APIs
+
+Proposed name of the module: `astro:manifest`
+
+Proposed sub-paths
+
+**By feature**:
+- `astro:manifest/routing`, exported bindings: 
+  - `i18n`
+  - `trailingSlash`
+  - `base`
+  - `build.format`
+  - `site` 
+  - `redirects`[^1]
+- `astro:manifest/fs`, exposed bindings: 
+  - `srcDirc`
+  - `cacheDir`
+  - `outDir`
+  - `build.client`
+  - `build.server`
+  - `build.serverEntry`
+  - `build.assetsPrefix`
+  - `publicDir`
+  - `root`
+
+**By "position"**
```

Comment:
I think I prefer by position, it will probably be clearer for users as it will kinda follow the config shape